src/lib/products.ts

```diff
@@ -18,6 +18,16 @@ const productSlugsToNames: { [slug in ProductSlug]: ProductName } = {
   vagrant: 'Vagrant',
 }
 
+/**
+ * Type guard to determine if a string is a ProductSlug
+ *
+ * TODO: should we define ProductSlug as an enum,
+ * so that we can use its values directly here?
+ */
+function isProductSlug(string: string): string is ProductSlug {
```

Comment:
This `isProductSlug` type guard is used in `pages/_error.tsx`'s `getServerSideProps` to ensure we're correctly dynamically detecting product context from URL segments.

Reply:
@zchsh and type guards, name a more dynamic duo 😛

Reply:
@BRKalow you converted me, I'm a TypeScript fan now ahah 🤣 I look back on my former disdain for TypeScript with a mix of guilt and nostalgia 🙇‍♂️ Next thing you know I'll be all-in working on Swingset 💖 😝